scripts/report_errors.sh

```diff
@@ -1,14 +1,9 @@
 #!/usr/bin/env bash
 
-export ERROR_LOG="/tmp/error.log"
+ERROR_LOG="$1"
 
 function report_error_to_github
 {
-    if [ $? -eq 0 ]
```

Comment:
Why not just `set -e` at the top?

Reply:
I think that the original reason might have been that this script used to be sourced by `check_style.sh` which uses grep a lot. grep returns 1 if there's no match.

But in this PR I wanted to include only the changes actually needed to fix the errors, without any extra refactors, to get it out today and minimize the the chance of introducing new problems. It's basically ready, I just need to remove the dummy commit with a style error.

I'm also preparing another PR with a more complete refactor of these two scripts (including fixes for shellcheck warnings from my pending draft PRs) and I'll try to use `set -e` there. But it's getting late so I think I'll finish that on Monday.

Reply:
Actually I decided to just post what I have in a separate PR: #10797. No `set -e` yet though :)

I'll go back to this later, when I'll be finalizing drafts of my other shellcheck-releated PRs (that are basically ready and just need some extra testing tomake sure they don't break stuff).

Reply:
@cameel there's a cyclic PR dependency here: you said above you want to do #10797  before this here. But #10797 says "NOTE: This PR is based on #10792 ", which in this one. So what's the order?

Reply:
This one (#10792) is first - it contains only the minimum necessary to fix the errors in CI.

#10797 is on top of it and contains only refactors that I managed to do on Friday. Unfortunately that does not include `set -e`.